src/TwigPageLoader.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+
+namespace WMDE\Fundraising\Frontend;
+
+use Twig_Error_Loader;
+use WMDE\Fundraising\Frontend\PageRetriever\PageRetriever;
+
+/**
+ * @license GNU GPL v2+
+ * @author Gabriel Birke < gabriel.birke@wikimedia.de >
+ */
+class TwigPageLoader implements \Twig_LoaderInterface {
+
+	private $pageRetriever;
+	private $pageCache = [];
+	private $errorCache = [];
+
+	/**
+	 * TwigPageLoader constructor.
+	 * @param $pageRetriever
+	 */
+	public function __construct( PageRetriever $pageRetriever ) {
+		$this->pageRetriever = $pageRetriever;
+	}
+
+	public function getSource( $name ) {
+		return $this->retrievePage( $name );
+	}
+
+	public function getCacheKey( $name ) {
+		// retrieve page to generate loader exception if page does not exist
+		$this->retrievePage( $name );
+		return $name;
+	}
+
+	public function isFresh( $name, $time ) {
```

Comment:
<darth vader voice>I find your lack of type hints disturbing</darth vader voice>